### PR TITLE
No longer add "has formula" to cells without one in Libre Office Calc 6.4

### LINKED ISSUE #10759: Libre Office Calc 6.4: NVDA reports "has formula" on each cell regardless if there is a formula
### Steps to reproduce:
1. Open NVDA
2. Open an empty table in Libre Office Calc 6.4
3. navigate with arrow keys

### Actual behavior:
NVDA reports "has formula" on every cell.

### Expected behavior:
NVDA should report "has formula" only when there is one formula in the cell.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2019.3 RC2

#### Windows version:
Windows 10 1909 update

#### Name and version of other software in use when reproducing the issue:
Libre Office Calc 6.4.

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
In 2019.1.1 I didn't see this issue.

### PR DESCRIPTION
**Pull request by lukaszgo1**
## Link to issue number:
Fixes #10759 
### Summary of the issue:
In #9115 we started announcing presence of Formulas In Libre Office calc. To detect them we are looking for the IAccessible 2 attribute Formula, and if it was not empty cell in question had formula inside. This worked well until Libre Office Calc started exposing "!!br0ken!!" as  content of this attribute when cell has no formula. For users 'formula' was announced with every cell.
### Description of how this pull request fixes the issue:
If the content of IAccessible 2 attribute is "!!br0ken!!" the STATE_HASFORMULA is not added.
I've considered doing  something more sophisticated to protect from similar breakages in the future - unfortunately the equals sign which denotes start of the formula is not a part of this attribute when there is a formula in the cell so this has to do.
### Testing performed:
1. In Libre Office Calc 6.3 ensured that formulas are reported as before.
2. In Libre Office Calc 6.4 created simple sheet with two cells containing numbers and one with a formula - for the one with numbers formula is not announced whereas for the one with formula it is.

### Known issues with pull request:
None
### Change log entry:

Bug fixes
- Formula is no longer announced for every cell in Libre Office Calc 6.4.

### CODE REVIEW

**Comment by michaelDCurran** [comment]
Just noting here (because it took me a long time to get it): the string represents "broken". 
I guess this pr makes sense, but I really really think an issue/pr should be opened against Calc as this is certainly not an appropriate usage of IAccessible2 attributes.

**Comment by lukaszgo1** [comment]
@michaelDCurran I've created [this bug report](https://bugs.documentfoundation.org/show_bug.cgi?id=132147).

**Comment by lukaszgo1** [comment]
@michaelDCurran Assuming you agree with what this PR does could you review and merge it? Without this change LO Calc 6.4 is very annoying to  use and the change is quite small - it would be bad if it would miss 2020.2

File: source/appModules/soffice.py [hunk 1]
```diff
@@ -220,8 +220,13 @@ def _get_states(self):
 			# #8988: Cells in Libre Office do not have the selected state when a single cell is selected (i.e. has focus).
 			# Since #8898, the negative selected state is announced for table cells with the selectable state.
 			states.add(controlTypes.STATE_SELECTED)
-		if self.IA2Attributes.get('Formula'):
-			# #860: Recent versions of Calc expose has formula state via IAccessible 2.
+		# #860: Recent versions of Calc expose has formula state via IAccessible 2.
+		# #10759: However in Libre Office 6.4 the attribute contains "!!br0ken!!"
+		# when there is no formula in the cell.
+		if (
+			self.IA2Attributes.get('Formula')
```
Comment (michaelDCurran) [inline source/appModules/soffice.py:227]: Could you fetch the value before the if check, storing it in a variable, and then use the variable in both checks please.

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

File: source/appModules/soffice.py (hunk 1 above)
Comment (lukaszgo1) [reply source/appModules/soffice.py]: @michaelDCurran Done

**Comment by codeofdusk** [comment]
Bug fixed upstream, I vote to close this.

**Comment by Adriani90** [comment]
Confirming this is fixed in Libre Office 6.4.4.2. Closing.

**Pull request closed without merging**